client/src/main/java/org/apache/rocketmq/client/impl/factory/MQClientInstance.java

```diff
@@ -255,19 +255,17 @@ public void start() throws MQClientException {
     }
 
     private void startScheduledTask() {
-        if (null == this.clientConfig.getNamesrvAddr()) {
```

Comment:
@Jaskey yes, this (current) design is surely ok. But it makes sense to discuss it in the dev mailing list, because everyone has a different idea what the best one is.

For instance, I would change what you describe a little bit like this
```
    1. when specified nameserver address, no need to fetch nameserver address dynamically.
    2. Only fetch nameserver address when the user-defined nameserver fails.
       I don't think any inconsistencies should happen, because it's a user's choice to enable dynamic fetching (We can implement such a switch to enable/disable it).
```